cloudinit/distros/netbsd.py

```diff
@@ -0,0 +1,231 @@
+# Copyright (C) 2019-2020 Gonéri Le Bouder
+#
+# This file is part of cloud-init. See LICENSE file for license information.
+
+import crypt
+import os
+import platform
+import six
+
+from cloudinit import distros
+from cloudinit import helpers
+from cloudinit import log as logging
+from cloudinit import net
+from cloudinit import ssh_util
+from cloudinit import util
+from cloudinit.distros import bsd_util
+
+LOG = logging.getLogger(__name__)
+
+
+class Distro(distros.Distro):
+    hostname_conf_fn = '/etc/rc.conf'
+    ci_sudoers_fn = '/usr/pkg/etc/sudoers.d/90-cloud-init-users'
+
+    def __init__(self, name, cfg, paths):
+        distros.Distro.__init__(self, name, cfg, paths)
+        # This will be used to restrict certain
+        # calls from repeatly happening (when they
+        # should only happen say once per instance...)
+        self._runner = helpers.Runners(paths)
+        self.osfamily = 'netbsd'
+        cfg['ssh_svcname'] = 'sshd'
+
+    def _read_system_hostname(self):
+        sys_hostname = self._read_hostname(filename='/etc/rc.conf')
+        return ('/etc/rc.conf', sys_hostname)
+
+    def _read_hostname(self, filename, default=None):
+        return bsd_util.get_rc_config_value('hostname')
+
+    def _write_hostname(self, hostname, filename):
+        bsd_util.set_rc_config_value('hostname', hostname,
+                                     fn='/etc/rc.conf')
+
+    def create_group(self, name, members):
```

Comment:
This looks a lot like the base class Distro.create_group;   what's different in here?  It appears to be the usermod command; so maybe in the base create_group you could make use of util.is_OpenBSD() ?

Reply:
i think @OddBloke suggested elsewhere that we have a variable for these commands, so we don't have to override the entire function, or clutter base with is_x()